File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ExtensionConfiguration.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class ExtensionConfiguration(
  val serverEndpoint: String? = null,
  val proxy: String? = null,
  val accessToken: String? = null,
  val customHeaders: Map<String, String>,
  val anonymousUserID: String? = null,
  val autocompleteAdvancedProvider: String? = null,
  val autocompleteAdvancedModel: String? = null,
  val debug: Boolean? = null,
  val verboseDebug: Boolean? = null,
  val telemetryClientName: String? = null,
  val codebase: String? = null,
  val customConfiguration: Map<String, Any>? = null,
  val customConfigurationJson: String? = null,
  val baseGlobalState: Map<String, Any>? = null,
)

File: agent/bindings/csharp/ExtensionConfiguration.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ExtensionConfiguration
  {
    [JsonProperty(PropertyName = "serverEndpoint")]
    public string ServerEndpoint { get; set; }
    [JsonProperty(PropertyName = "proxy")]
    public string Proxy { get; set; }
    [JsonProperty(PropertyName = "accessToken")]
    public string AccessToken { get; set; }
    [JsonProperty(PropertyName = "customHeaders")]
    public Dictionary<string, string> CustomHeaders { get; set; }
    [JsonProperty(PropertyName = "anonymousUserID")]
    public string AnonymousUserID { get; set; }
    [JsonProperty(PropertyName = "autocompleteAdvancedProvider")]
    public string AutocompleteAdvancedProvider { get; set; }
    [JsonProperty(PropertyName = "autocompleteAdvancedModel")]
    public string AutocompleteAdvancedModel { get; set; }
    [JsonProperty(PropertyName = "debug")]
    public bool Debug { get; set; }
    [JsonProperty(PropertyName = "verboseDebug")]
    public bool VerboseDebug { get; set; }
    [JsonProperty(PropertyName = "telemetryClientName")]
    public string TelemetryClientName { get; set; }
    [JsonProperty(PropertyName = "codebase")]
    public string Codebase { get; set; }
    [JsonProperty(PropertyName = "eventProperties")]
    public EventProperties EventProperties { get; set; }
    [JsonProperty(PropertyName = "customConfiguration")]
    public Dictionary<string, Object> CustomConfiguration { get; set; }
    [JsonProperty(PropertyName = "baseGlobalState")]
    public Dictionary<string, Object> BaseGlobalState { get; set; }
  }
}
